### PR TITLE
Stats website in vue3

### PR DESCRIPTION
**Pull request by 0oM4R**
### Description
New stats website using Vue3 
this website is similar to the old stats website but i will mention some of the new features 
- support standby nodes stats
- network filter 
  - The user can select one or more networks to get their stats
    Screenshot from 2023-10-23 17-40-30
    - if the user tried to switch off all toggles will got a warning dialog
      Screenshot from 2023-10-24 11-22-12

    - the website will make a request to get a network stats only if the user click on refresh button and on mount
    
 - responsive 
     image
- loading error  
 
image

   

### Changes

List of changes this PR includes

### Related Issues

- #467 
- #1275

### Checklist

- [ ] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/new_stats/src/components/stats_table.vue [hunk 1]
```diff
@@ -0,0 +1,132 @@
+<template>
+  <div>
+    <!--loading-->
+    <v-sheet color="transparent" height="82vh" v-if="loading" class="d-flex align-center justify-center">
+      <v-container class="text-center d-block">
+        <v-progress-circular size="40" indeterminate />
+        <p class="pt-4 font-weight-bold">Loading stats data</p>
```
Comment (maayarosama) [inline packages/new_stats/src/components/stats_table.vue:7]: Loading Statistics

**Comment by zaelgohary** [comment]
Can we change the card size in the responsive view. It's not normal for the card to be bigger than the map. You can check with @ehab-hassan for a better view.

image

File: packages/new_stats/index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <script defer src="https://cdn.jsdelivr.net/gh/threefoldtech/tf-map@0.0.4/dist/tf-map.js"></script>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Vite App</title>
```
[the remark this answers is not in the document]
Comment (0oM4R) [reply packages/new_stats/index.html]: image

**Review by maayarosama — APPROVED** [review]

File: packages/new_stats/src/components/stats_table.vue [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      { data: formattedStats.value.accessNodes, title: "Access Nodes", icon: "mdi-gate" },
+      { data: formattedStats.value.gateways, title: "Gateways", icon: "mdi-boom-gate-outline" },
+      { data: formattedStats.value.twins, title: "Twins", icon: "mdi-brain" },
+      { data: formattedStats.value.publicIps, title: "Public IPs", icon: "mdi-access-point" },
+      { data: formattedStats.value.contracts, title: "Contracts", icon: "mdi-file-document-edit-outline" },
+    ];
+  }
+});
+async function getStatsData(refresh = false) {
+  if (props.networks) {
```
Comment (zaelgohary) [inline packages/new_stats/src/components/stats_table.vue:102]: I don't think we need to condition this as you're already calling the function if props.networks changed in line 118.

File: packages/new_stats/src/components/stats_table.vue [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    </v-sheet>
+
+    <!--error-->
+    <v-sheet color="transparent" height="82vh" v-else-if="failed" class="d-flex align-center w-100 justify-center">
+      <v-container class="text-center">
+        <v-icon color="error" size="x-large">mdi-close-circle-outline</v-icon>
+        <v-container class="text-error">
+          Failed to get stats data, Please check you internet connection or try again later</v-container
+        >
+        <v-btn class="text-capitalize" @click="getStatsData(true)" color="secondary">Try again </v-btn>
```
Comment (zaelgohary) [inline packages/new_stats/src/components/stats_table.vue:18]: Add disabled attribute to the button if props.networks haven't been passed yet.

File: packages/new_stats/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+## Introduction
+
+Grid Stats is a web application created to replace explorer.threefold.io with a simple page showing a capacity map + statistics. In the map, you get to see node distribution with each node location highlighted. While in Statistics section, general details about the nodes are being displayed like online nodes, farms count, total storage used, gateways ...etc.
+Also, you can select one or more networks to get their stats.
+
+## Installation
+
+- **Prerequisite**
+
+  - Nodejs 16^
```
Comment (zaelgohary) [inline packages/new_stats/README.md:12]: Some SDK packages require Nodejs 18, so I think it would be better if we used 18^ as it's in the same repo.

**Review by zaelgohary — COMMENTED** [review]
Nice work. Some comments on packages that are not used like cypress & vitest. There aren't any tests written for those so I don't think both are necessary for atm.

File: packages/new_stats/package.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "newstats",
```
[the remark this answers is not in the document]
Comment (0oM4R) [reply packages/new_stats/package.json:2]: we keep this name till we phase out the old stats

File: packages/new_stats/README.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+```bash
+cd tfgrid-sdk-ts
+yarn install
+```
+
+You can now run the project in development mode using
+
+```bash
+yarn workspace newstats dev
```
[the remark this answers is not in the document]
Comment (0oM4R) [reply packages/new_stats/README.md:34]: will keep this name till we phase out the old stats

File: packages/new_stats/src/components/stats_table.vue (hunk 3 above)
Comment (0oM4R) [reply packages/new_stats/src/components/stats_table.vue:102]: It was for the initial value; it was null but currently we are sending the init value with all networks;
good catch thank you !

File: packages/new_stats/src/components/stats_table.vue (hunk 4 above)
Comment (0oM4R) [reply packages/new_stats/src/components/stats_table.vue:18]: there is no way for this to happen, the initial stat of networks [dev,test,main] and the user can't switch off more than two toggles

File: packages/new_stats/package.json [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "dev": "vite",
+    "build": "run-p type-check \"build-only {@}\" --",
+    "preview": "vite preview",
+    "test:unit": "vitest",
```
Comment (zaelgohary) [inline packages/new_stats/package.json:9]: Shouldn't we also remove this? and all Vitest commands and files.

File: packages/new_stats/vitest.config.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { fileURLToPath } from "node:url";
```
Comment (zaelgohary) [inline packages/new_stats/vitest.config.ts:1]: This should be removed.

File: packages/new_stats/tsconfig.vitest.json [hunk 10]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (zaelgohary) [inline packages/new_stats/tsconfig.vitest.json:1]: Also this.

**Review by zaelgohary — CHANGES_REQUESTED** [review]

File: packages/new_stats/package.json (hunk 8 above)
Comment (zaelgohary) [reply packages/new_stats/package.json:9]: No need to remove Vitest as it's used in type checking.

**Review by zaelgohary — APPROVED** [review]

**Review by maayarosama — APPROVED** [review]

**Pull request merged**